### PR TITLE
DEVPROD-6861: Gate permissions for Task.taskLogs in GraphQL

### PR DESCRIPTION
**Pull request by SupaJoon**
DEVPROD-6861

### Description
These code changes allow the user to request `Task.taskLogs` if they have the LogsView role for the project.

### CODE REVIEW

File: graphql/util.go [hunk 1]
```diff
@@ -1310,6 +1301,34 @@ func hasAnnotationPermission(ctx context.Context, obj *restModel.APITask, requir
...
+	permissions := gimlet.PermissionOpts{
+		Resource:      *obj.ProjectId,
+		ResourceType:  evergreen.ProjectResourceType,
+		Permission:    evergreen.PermissionLogs,
+		RequiredLevel: evergreen.LogsView.Value,
+	}
+	if authUser.HasPermission(permissions) {
+		return true, nil
+	}
+	return isPatchAuthorForTask(ctx, obj)
```
Comment (khelif96) [inline graphql/util.go:1315]: Is the implication here that patch authors can always view their own logs even if they don't have permission to view other logs in the project? This seems like a security loophole.

Comment (SupaJoon) [reply graphql/util.go:1315]: That's right. This is what we do for task annotations. The user needs the [Patch Submit](https://github.com/evergreen-ci/evergreen/blob/f9f7299af25bd7a65750c0fd20a0c8b6bc37ff40/globals.go#L1258) permission to submit a task. Sometimes we grant bots the ability to submit patches, maybe this is a potential loophole.

**Review by khelif96 — APPROVED** [review]
Maybe someone from some app should also review the security logic to see if it makes sense and conforms with the way its supposed to be handled.

**Review by Kimchelly — COMMENTED** [review]
* Did we test the permissions changes in staging?
* I don't have particular opinions on whether users should be able to view their own patch annotations without the explicit permission. But since the patch special case is only there for annotations and isn't relevant to task logs, it doesn't seem relevant to the ticket. I think we should just make a separate ticket to investigate the annotation special case.

**Review by Kimchelly — APPROVED** [review]

**Pull request merged**